aretext/aretext
# Add relative line number

Description
-----------

- Add configuration option "lineNumberMode"
- Test the display of line numbers
- Set default to "absolute"
- Add documentation in configuration reference
- Add command to toggle line number mode to "nur" (show line number is "nu")

I tried my best to be a good citizen. Please comment on whatever seems inadequate as I don't have strong opinion on most things, like providing a menu command for that. The thinking behind it is that if it is in the config, it is nice to have it as menu command.
Let me know your thoughts. 

My only personal preference is to replace testity but it is very much stylistic and doesn't bear more weight than to say "one less dependency". See an example of a simple test framework [here](https://github.com/volvo-cars/lingon/blob/main/pkg/testutil/assert.go). 

I have to say that this repo is very well maintained and it took very little time to get familiar with the code base. Congratulations on that, it was a delight to contribute. 

Issue
-----

implements #176 

Checklist
---------

- [x] I have read the [Contribution Guidelines](https://github.com/aretext/aretext/blob/main/CONTRIBUTING.md)
- [x] I have added tests for new features or bug fixes in this PR.
- [x] I have updated any related documentation (markdown files in the "docs" directory).
- [x] I have run `make` and checked in any changes to generated code.
- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- [x] I have formatted all commit messages in the same style as [the other commits in the repository](https://github.com/aretext/aretext/commits/).
- [x] I have added a "Signed-off-by" trailer in all commit messages (`git commit -s`) to indicate my agreement with the [Developer Certificate of Origin](https://developercertificate.org/).
